From d25e78f286578b4d3cf7e279c990986801262c3d Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 28 Aug 2020 13:01:28 -0400
Subject: [PATCH] python: Fix libpyuhd install_name on macOS.

---
 host/python/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/host/python/CMakeLists.txt b/host/python/CMakeLists.txt
index 04e637f2e..1cd8b59ec 100644
--- a/host/python/CMakeLists.txt
+++ b/host/python/CMakeLists.txt
@@ -116,6 +116,11 @@ else()
         DESTINATION ${UHD_PYTHON_DIR}
         COMPONENT pythonapi
     )
+    if(APPLE)
+        # CMAKE_INSTALL_NAME_DIR is set to ${CMAKE_INSTALL_PREFIX}/${LIBRARY_DIR},
+        # but this is wrong for pyuhd and needs to be replaced with the correct dir
+        set_target_properties(pyuhd PROPERTIES INSTALL_NAME_DIR ${UHD_PYTHON_DIR}/uhd)
+    endif(APPLE)
     # On Linux/Unix systems, we must properly install the library file, though.
     # install(DIRECTORY) will treat the .so file like any other file, which
     # means it won't update its RPATH, and thus the RPATH would be stuck to the
-- 
2.20.1

