From 8f8442cf41ddbba69d8934fcb08345d0810c3266 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 28 Aug 2020 11:09:48 -0400
Subject: [PATCH] python: Update virtualenv check to work with venv and conda.

---
 host/python/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/host/python/CMakeLists.txt b/host/python/CMakeLists.txt
index 04e637f2e..dc226b3b8 100644
--- a/host/python/CMakeLists.txt
+++ b/host/python/CMakeLists.txt
@@ -10,7 +10,7 @@
 
 PYTHON_CHECK_MODULE(
     "virtualenv"
-    "sys" "hasattr(sys, 'real_prefix')"
+    "os, sys" "hasattr(sys, 'real_prefix') or (hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix) or os.environ['CONDA_DEFAULT_ENV']"
     HAVE_PYTHON_VIRTUALENV
 )
 
-- 
2.20.1

