From 4f23b68e95eec411289e708b50a8b8368db941d3 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 28 Aug 2020 13:01:28 -0400
Subject: [PATCH 02/10] python: Fix libpyuhd install_name on macOS.

---
 host/cmake/Modules/UHDPython.cmake | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/host/cmake/Modules/UHDPython.cmake b/host/cmake/Modules/UHDPython.cmake
index ebb6b1a..81344b1 100644
--- a/host/cmake/Modules/UHDPython.cmake
+++ b/host/cmake/Modules/UHDPython.cmake
@@ -299,6 +299,14 @@ macro(UHD_PYTHON_INSTALL_MODULE)
                 DESTINATION ${UHD_PYTHON_DIR}
                 COMPONENT pythonapi
             )
+            if(APPLE)
+                # CMAKE_INSTALL_NAME_DIR is set to
+                # ${CMAKE_INSTALL_PREFIX}/${LIBRARY_DIR}, but this is wrong for
+                # pyuhd and needs to be replaced with the correct dir
+                # First, get the real path by following any symlinks
+                get_filename_component(PYUHD_NAME_DIR ${UHD_PYTHON_DIR}/uhd REALPATH)
+                set_target_properties(pyuhd PROPERTIES INSTALL_NAME_DIR ${PYUHD_NAME_DIR})
+            endif(APPLE)
             # On Linux/Unix systems, we must properly install the library file.
             # install(DIRECTORY) will treat the .so file like any other file, which
             # means it won't update its RPATH, and thus the RPATH would be stuck to the
