From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 28 Aug 2020 13:01:28 -0400
Subject: [PATCH] python: Fix libpyuhd install_name on macOS.

---
 host/python/CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/host/python/CMakeLists.txt b/host/python/CMakeLists.txt
index 212bd9e46..c5de3c065 100644
--- a/host/python/CMakeLists.txt
+++ b/host/python/CMakeLists.txt
@@ -191,6 +191,13 @@ else()
         DESTINATION ${UHD_PYTHON_DIR}
         COMPONENT pythonapi
     )
+    if(APPLE)
+        # CMAKE_INSTALL_NAME_DIR is set to ${CMAKE_INSTALL_PREFIX}/${LIBRARY_DIR},
+        # but this is wrong for pyuhd and needs to be replaced with the correct dir
+        # First, get the real path by following any symlinks
+        get_filename_component(PYUHD_NAME_DIR ${UHD_PYTHON_DIR}/uhd REALPATH)
+        set_target_properties(pyuhd PROPERTIES INSTALL_NAME_DIR ${PYUHD_NAME_DIR})
+    endif(APPLE)
     # On Linux/Unix systems, we must properly install the library file, though.
     # install(DIRECTORY) will treat the .so file like any other file, which
     # means it won't update its RPATH, and thus the RPATH would be stuck to the
-- 
2.35.1

