From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 6 Dec 2024 12:01:59 -0500
Subject: [PATCH] cmake: Remove dynamic_lookup, flat_namespace linker flags for
 Apple

---
 host/lib/CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/host/lib/CMakeLists.txt b/host/lib/CMakeLists.txt
index f5b9f1609..e9d29a6f6 100644
--- a/host/lib/CMakeLists.txt
+++ b/host/lib/CMakeLists.txt
@@ -182,10 +182,6 @@ if(ENABLE_MPMD)
 endif()
 add_library(uhd SHARED ${libuhd_sources})
 target_link_libraries(uhd PUBLIC ${Boost_LIBRARIES} ${libuhd_libs})
-if(APPLE)
-    target_link_options(uhd PRIVATE "LINKER:-undefined,dynamic_lookup")
-    target_link_options(uhd PRIVATE "-flat_namespace")
-endif(APPLE)
 set_target_properties(uhd PROPERTIES DEFINE_SYMBOL "UHD_DLL_EXPORTS")
 if(NOT LIBUHDDEV_PKG)
     set_target_properties(uhd PROPERTIES SOVERSION "${UHD_ABI_VERSION}")
-- 
2.46.0

